src/Nest/CommonAbstractions/SerializationBehavior/StatefulDeserialization/ConcreteTypeConverter.cs

```diff
@@ -94,7 +94,10 @@ internal static class ConcreteTypeConverter
 
 		private static JObject CreateIntermediateJObject(JsonReader reader)
 		{
+			var original = reader.DateParseHandling;
+			reader.DateParseHandling = DateParseHandling.None;
```

Comment:
CovariantSearch needs ConcreteTypeConverter, in order to create an isolated serializer that has access to state we call the Stateful deserialization factory on `ISerializerFactory` and we append a new instance of ConcreteTypeConverter. Search/MultiGet/MultiSearch/SearchTemplate all possibly call this (but only if needed)